add_gist.rb

```diff
@@ -0,0 +1,74 @@
+require 'rubygems'
+require 'net/http'
+require 'uri'
+require 'json'
+require 'OpenSSL'
+require 'find'
+require 'dotenv/load'
+
+module AddGist
+  GISTS_URL = 'https://api.github.com/gists'.freeze
+
+  def self.upload_files(path, options = {})
+    files = read_files(path)
+    return if files.empty?
+
+    response = send_request(files, options)
+
+    if response.is_a?(Net::HTTPCreated)
+      html_url = JSON.parse(response.body)['html_url']
+      puts "Gist created successfully! \nAccess URL: #{html_url}"
+    else
+      puts response.code
+    end
+  end
+
+  def self.read_files(pwd = Dir.pwd)
+    files = {}
+    Find.find(pwd) do |path|
+      files.merge!(File.basename(path) => { 'content': File.open(path, 'r+').read }) unless File.directory?(path)
+    end
+    files
+  rescue Errno::ENOENT => e
+    puts e.message
```

Comment:
considering that eventually, the code could end here, I would add something to the error message besides the error that was raised.